src/services/globalThisShim.ts

```diff
@@ -0,0 +1,50 @@
+// We polyfill `globalThis` here so re can reliably patch the global scope
+// in the contexts we want to in the same way across script and module formats
+
+// https://mathiasbynens.be/notes/globalthis
+
+// #region The polyfill starts here.
+((() => {
+    if (typeof globalThis === "object") return;
+    try {
+        Object.defineProperty(Object.prototype, "__magic__", {
+            get() {
+                return this;
+            },
+            configurable: true
+        });
+        //@ts-ignore
+        __magic__.globalThis = __magic__;
+        // The previous line should have made `globalThis` globally
+        // available, but it fails in Internet Explorer 10 and older.
+        // Detect this failure and fall back.
+        if (typeof globalThis === "undefined") {
+            // Assume `window` exists.
+            //@ts-ignore
+            window.globalThis = window;
+        }
+        //@ts-ignore
+        delete Object.prototype.__magic__;
+    }
+    catch (error) {
+        // In IE8, Object.defineProperty only works on DOM objects.
+        // If we hit this code path, assume `window` exists.
+        //@ts-ignore
+        window.globalThis = window;
+    }
+})());
+// #endregion The polyfill ends here.
+
+// if `process` is undefined, we're probably not running in node - patch legacy members onto the global scope
+// @ts-ignore
+if (typeof process === "undefined" || process.browser) {
+    /// TODO: this is used by VS, clean this up on both sides of the interface
+    //@ts-ignore

```

Comment:
mixed feelings about ts-ignore in ts file in Typescript codebase...

Reply:
Eh, we're doing things that are untyped here (and the types "fix" is actually declaring these fields as globals, which I'd rather not do), so we need the bailout. It's also not our first - we already have one in the mapShim file to suppress an "unused type parameter" error (and is needed because said interface later merges with another in a dependent project where the param is actually used).